api/app/Http/Controllers/UserController.php

```diff
@@ -28,7 +50,81 @@ class UserController extends BaseController
      */
     public function myAccount(Request $request) : JsonResponse
     {
-        $user = $request->input("currentUser");
-        return response()->json(UserDTOMapper::mapUser($user), 200);
+        $currentUser = $request->input("currentUser");
+        return response()->json(UserDTOMapper::mapModelToUserDTO($currentUser), 200);
+    }
+    
+    /**
+     * GET /users
+     *
+     * Get current user info.
+     *
+     * @param $request Request request.
+     * @return JsonResponse current users.
+     * @throws Exception
+     */
+    public function index(Request $request): JsonResponse
+    {
+        /** @var User $currentUser */
+        $currentUser = $request->input("currentUser");
+        
+        if ($currentUser->getRoleId() !== $this->roles->Admin) {
+            return response()->json(null, 403);
+        }
+        
+        $users = $this->userRepository->getUsers();
+        
+        $usersDTOs = array_map(function (User $user) {
+            return UserDTOMapper::mapModelToUserDTO($user);
+        },
+        $users);
+        
+        return response()->json($usersDTOs, 200);
+    }
+    
+    public function update(Request $request, $id): JsonResponse
+    {
+        /** @var User $currentUser */
+        $currentUser = $request->input("currentUser");
+        
+        if (!(($currentUser->getUserId() === (int)$id) ||
+            ($currentUser->getRoleId() === $this->roles->Admin)))
+        {
+            return response()->json(null, 403);
+        }
+    
+        try {
+            $userDTO = $this->jsonMapper->map($request->json(), new UserDTO());
+        } catch (Exception $e) {
+            return response()->json(null, 400);
+        }
+    
+        $user = UserDTOMapper::mapUserDTOToModel($userDTO);
+    
+        $result = $this->userRepository->updateUser($id, $user);
+        if ($result) {
+            return response()->json(null, 200);
+        } else {
+            return response()->json(null, 500);
+        }
+    }
+    
+    public function refreshToken(Request $request, $id): JsonResponse
+    {
+        /** @var User $currentUser */
+        $currentUser = $request->input("currentUser");
+        
+        if ($currentUser->getRoleId() !== $this->roles->Admin) {
+            return response()->json(null, 403);
+        }
+        
+        $newApiKey = md5(Guid::uuid4()->toString());
+        
+        $result = $this->userRepository->updateApiKey($id, $newApiKey);
+        if (!!$result) {
+            return response()->json($newApiKey, 200);
+        } else {
+            return response()->json(null, 500);
```

Comment:
You should not ever return 500. This status states that you failed to process some behavior
Why null? At least something meaningful